mne/utils/check.py

```diff
@@ -330,3 +330,69 @@ def _check_if_nan(data, msg=" to be plotted"):
     """Raise if any of the values are NaN."""
     if not np.isfinite(data).all():
         raise ValueError("Some of the values {} are NaN.".format(msg))
+
+
+def _check_info_inv(info, forward, data_cov=None, noise_cov=None):
+    """Return good channels common to forward model and covariance matrices."""
+    from .. import pick_types
+    # get a list of all channel names:
+    fwd_ch_names = forward['info']['ch_names']
+
+    # handle channels from forward model and info:
+    ch_names = _compare_ch_names(info['ch_names'], fwd_ch_names, info['bads'])
+
+    # make sure that no reference channels are left:
+    ref_chs = pick_types(info, meg=False, ref_meg=True)
+    ref_chs = [info['ch_names'][ch] for ch in ref_chs]
+    ch_names = [ch for ch in ch_names if ch not in ref_chs]
+
```

Comment:
@agramfort : this code block is new as well, throwing out any reference channels that might still be in ``info`` for the beamformer. Now that we have this function in ``utils`` we probably need to reconsider whether this behavior is wanted by any inverse model?

Reply:
IIRC the forward does not contain the reference channels (only implicitly via compensation if it has been applied when computing the forward) so I don't think it should matter.

Reply:
Yes, that is true, I acutally checked that. Talking to @agramfort, we thought about getting rid of them to ensure further computation doesn't go wrong in any case for the beamformer.